### PR TITLE
Add entire album to queue when in album shuffle mode.

### PR DESCRIPTION
**Pull request by stephenjsweeney**

### CODE REVIEW

File: src/ch/blinkenlights/android/vanilla/SongTimeline.java [hunk 1]
```diff
@@ -580,10 +582,18 @@ public Song getSong(int delta)
...
+					if (songs.size() == 0) {
 						return null;
-					timeline.add(song);
+					}
+
+					for (Song newSong : songs) {
+						timeline.add(newSong);
+					}
+
+					song = songs.get(songs.size() - 1);
```
Comment (adrian-bl) [inline src/ch/blinkenlights/android/vanilla/SongTimeline.java]: Shouldn't this be `songs.get(0)` ?
The current code returns the last song we added while the caller just asked for the next song (which would be the first in our list)

File: src/ch/blinkenlights/android/vanilla/MediaUtils.java [hunk 2]
```diff
@@ -426,12 +428,36 @@ public static Song getRandomSong(Context context, boolean albumShuffle) {
...
+			// since we're in album shuffle mode, we'll want to add in the entire album in one go,
+			// so loop through the upcoming songs and add all those that have the same album id
+			// as the song we initially got
+			if (songs.size() > 0 && albumShuffle) {
+				long expectedAlbumId = song.albumId;
+				boolean hasMore = true;
+
+				while (hasMore) {
+					hasMore = false;
+					if (songs.size() > 0 && songs.get(0).albumId == expectedAlbumId) {
```
Comment (adrian-bl) [inline src/ch/blinkenlights/android/vanilla/MediaUtils.java]: This looks clunky :-)

How about a `do {} while()` loop?

File: src/ch/blinkenlights/android/vanilla/MediaUtils.java [hunk 3]
```diff
@@ -426,12 +428,36 @@ public static Song getRandomSong(Context context, boolean albumShuffle) {
...
-		Song result = null;
+		final List<Song> results = new ArrayList<>();
+
 		if (songs.size() > 0) {
-			result = songs.remove(0);
-			result.flags |= Song.FLAG_RANDOM;
+			final Song song = songs.remove(0);
+
+			if (!albumShuffle) {
+				song.flags |= Song.FLAG_RANDOM;
```
Comment (adrian-bl) [inline src/ch/blinkenlights/android/vanilla/MediaUtils.java]: The current logic is to remove auto-added songs if the user disables the random mode or adds new songs.
Guess it's okay to keep albums in the queue (even if they were auto added), but maybe add a comment about this.

Comment (stephenjsweeney) [reply src/ch/blinkenlights/android/vanilla/MediaUtils.java]: Done so. I could investigate in future popping in a new flag to say the song is from a random album and allow it to be removed if the album is not currently playing?

File: src/ch/blinkenlights/android/vanilla/MediaUtils.java (hunk 2 above)
Comment (stephenjsweeney) [reply src/ch/blinkenlights/android/vanilla/MediaUtils.java]: Yep. Changed to a do...while{} loop.

File: src/ch/blinkenlights/android/vanilla/SongTimeline.java (hunk 1 above)
Comment (stephenjsweeney) [reply src/ch/blinkenlights/android/vanilla/SongTimeline.java]: Changed back to getting 0.

**Comment by adrian-bl** [comment]
Looks fine to me: thanks for the pull request!

**Pull request merged**